AdyenComponents/Apple Pay/ApplePayComponentExtensions.swift

```diff
@@ -34,14 +34,96 @@ extension ApplePayComponent: PKPaymentAuthorizationViewControllerDelegate {
+                                      shippingMethod: payment.shippingMethod)
         
         submit(data: PaymentComponentData(paymentMethodDetails: details, amount: self.amountToPay, order: order))
     }
+
+    public func paymentAuthorizationViewController(
+        _ controller: PKPaymentAuthorizationViewController,
+        didSelectShippingContact contact: PKContact,
+        handler completion: @escaping (PKPaymentRequestShippingContactUpdate) -> Void
+    ) {
+        guard let applePayDelegate = applePayDelegate else {
+            return completion(.init(paymentSummaryItems: applePayPayment.summaryItems))
+        }
+
+        let result = applePayDelegate.didUpdate(contact: contact,
+                                                with: applePayPayment,
+                                                component: self)
+        if result.paymentSummaryItems.isEmpty == false {
+            do {
+                try applePayPayment.update(with: result.paymentSummaryItems)
+            } catch {
+                delegate?.didFail(with: error, from: self)
+            }
+        }
+        print(applePayPayment.summaryItems.reduce("") { $0 + "| \($1.label): \($1.amount.floatValue.rounded()) " })
+        completion(result)
+    }
+
+    public func paymentAuthorizationViewController(_ controller: PKPaymentAuthorizationViewController,
+                                                   didSelect shippingMethod: PKShippingMethod,
+                                                   handler completion: @escaping (PKPaymentRequestShippingMethodUpdate) -> Void) {
+        guard let applePayDelegate = applePayDelegate else {
+            return completion(.init(paymentSummaryItems: applePayPayment.summaryItems))
+        }
+
+        let result = applePayDelegate.didUpdate(shippingMethod: shippingMethod,
+                                                with: applePayPayment,
+                                                component: self)
+        if result.paymentSummaryItems.isEmpty == false {
+            do {
+                try applePayPayment.update(with: result.paymentSummaryItems)
+            } catch {
+                delegate?.didFail(with: error, from: self)
+            }
+        }
+        print(applePayPayment.summaryItems.reduce("") { $0 + "| \($1.label): \($1.amount.floatValue.rounded()) " })
+        completion(result)
+    }
+
+    @available(iOS 15.0, *)
+    public func paymentAuthorizationViewController(_ controller: PKPaymentAuthorizationViewController,
+                                                   didChangeCouponCode couponCode: String,
+                                                   handler completion: @escaping (PKPaymentRequestCouponCodeUpdate) -> Void) {
+        guard let applePayDelegate = applePayDelegate else {
+            return completion(.init(paymentSummaryItems: applePayPayment.summaryItems))
+        }
+
+        let result = applePayDelegate.didUpdate(couponCode: couponCode,
```

Comment:
maybe those delegate functions need to be asynchronous as well?

Reply:
Good point 🤔